scripts/legacy_expression_tool.md

```diff
@@ -0,0 +1,171 @@
+]
+```
+
+### `objects_replaced.json`
+
+This file contains a list of objects where the legacy expressions have been replaced with all the information associated. This 
+file is useful to preview the changes expected to be done in the database before applying them.
+
+### `objects_backup.json`
+
+This file contains a list of objects containing legacy expression that have been backed up. This file is useful to preview the 
+changes expected to be done in the database before modifying it.
+
+## Command line arguments
+
+The script can be executed using the following command:
+
+```bash
+python legacy_expression_tool.py \
+    --host <mongodb-host> \
+    --port <mongodb-port> \
+    --database <mongodb-db> \
+    --collection <mongodb-collection> \
+    --translation <translation-file> 
+```
+
+The list of possible arguments that the scripts accepts are:
+
+| Argument | Description | Default value | Mandatory |
+| -------- | ----------- | ------------- | --------- |
+| `-H`, `--host` | The host of the MongoDB instance | `localhost` | No |
+| `-p`, `--port` | The port of the MongoDB instance | `27017` | No |
+| `-db`, `--database` | The database name to replace the expressions | NA | Yes |
+| `-c`, `--collection` | The collection name to replace the expressions | NA | Yes |
+| `-t`, `--translation` | The translation dictionary file to replace the expressions | `translation.json` | No |
+| `-D`, `--debug` | Enable debug mode | `False` | No |
+| `-cm`, `--commit` | Commit the changes to the database | `False` | No |
+| `-el`, `--expression-language` | What to do with the expression language field. Possibles values: `delete`, `ignore` or `jexl` | `ignore` | No |
+| `-stat`, `--statistics` | Print match statistics. Aggregation modes are the possible values: `service` and `subservice` | `service` | No |
+| `-S`, `--service` | The fiware service filter to replace the expressions | All subservices | No |
+| `-P`, `--service-path` | The fiware service path filter to replace the expressions | All subservices | No |
+| `-i`, `--deviceid` | The device id filter to replace the expressions | All devices | No |
+
+## Usage
+
+### Getting legacy expressions matches  
+
+The recommended way to use the script is to execute it in first stage without passing the commit parameter. This will generate the file `legacy_expression_list.json` containing all the legacy expressions found in the database. Then, you can use this file to build the translation file.
+
+```bash
+python legacy_expression_tool.py \
+    --host <mongodb-host> \
+    --port <mongodb-port> \
+    --database <mongodb-db> \
+    --collection <mongodb-collection>
+```
+This will generate 4 files. At this point, the most relevants are `legacy_expression_list.json` and `legacy_expression_ocurrences.json`. The first one contains a list of all the legacy expressions found in the database. The second one contains information about the objects where the legacy expressions are found.
```

Comment:
4 files or "up to 4 files". My doubt is regarding objects_backup.json. Is always generated? Or only when some document is effectively modified?

Reply:
In additon, legacy_expression_list.json and legacy_expression_occurencense.json could be links to the sections of this document where they are described.

Reply:
4 files are always generated each time the script is executed

Links Added in https://github.com/telefonicaid/iotagent-node-lib/pull/1381/commits/cc9693fe782f6da453121ccdbeca6fad9b4f3997